plugins/SitesManager/Controller.php

```diff
@@ -335,4 +338,69 @@ private function getVueInitializeCode($vueVersion = '3')
 INST;
     }
+
+    private function mergeMultipleNotification(&$templateData)
+    {
+        $isNotificationsMerged = false;
+        $bannerMessage = '';
+        $guides = [];
+        $message = [];
+
+        if ($templateData['ga3Used'] || $templateData['ga4Used']) {
+            $message[0] = 'Google Analytics ';
+            $ga3GuideUrl =  '<a href="https://matomo.org/faq/how-to/migrate-from-google-analytics-3-to-matomo/" target="_blank" rel="noreferrer noopener">Google Analytics 3</a>';
+            $ga4GuideUrl =  '<a href="https://matomo.org/faq/how-to/migrate-from-google-analytics-4-to-matomo/" target="_blank" rel="noreferrer noopener">Google Analytics 4</a>';
+            if ($templateData['ga3Used'] && $templateData['ga4Used']) {
+                $isNotificationsMerged = true;
+                $guides[] = $ga3GuideUrl;
+                $guides[] = $ga4GuideUrl;
+                $message[0] .= '3 & 4';
+            } else {
+                $message[0] .= ($templateData['ga3Used'] ? 3 : 4);
+                $guides[] = ($templateData['ga3Used'] ? $ga3GuideUrl : $ga4GuideUrl);
+            }
+        }
+
+        if (!empty($message) && $templateData['consentManagerName']) {
+            $isNotificationsMerged = true;
+            $message[] = $templateData['consentManagerName'];
+            $guides[] =  '<a href="' . $templateData['consentManagerUrl'] . '" target="_blank" rel="noreferrer noopener">' . $templateData['consentManagerName'] . '</a>';
+        }
+
+        if (!empty($message)) {
+            $bannerMessage = StaticContainer::get(Translator::class)->createAndListing($message);
+        }
+
+        if ($isNotificationsMerged && $bannerMessage) {
+            $info = [
+                'isNotificationsMerged' => $isNotificationsMerged,
+                'notificationMergedMessage' => '<p class="fw-bold">' . Piwik::translate('SitesManager_MergedNotificationLine1', [$bannerMessage]) . '</p><p>' . Piwik::translate('SitesManager_MergedNotificationLine2', [(implode(' / ', $guides))]) . '</p>'
+            ];
+
+            if (!empty($templateData['consentManagerIsConnected'])) {
+                $info['notificationMergedMessage'] .= '<p>' . Piwik::translate('SitesManager_ConsentManagerConnected', [$templateData['consentManagerName']]) . '</p>';
+            }
+        } else {
+            $info = $this->getSingleNotifications($templateData);
+        }
+
+        $templateData = array_merge($templateData, $info);
+    }
+
+    private function getSingleNotifications(&$templateData)
+    {
+        $info = ['isNotificationsMerged' => false, 'notificationMergedMessage' => ''];
+        if (!empty($templateData['consentManagerName']) ) {
+            $info['notificationMergedMessage'] = '<p>' . Piwik::translate('PrivacyManager_ConsentManagerDetected', [$templateData['consentManagerName'], '<a href="' . $templateData['consentManagerUrl'] . '" target="_blank" rel="noreferrer noopener">', '</a>']) . '</p>';
```

Comment:
This can now be `notificationMessage` everywhere since there's always only one, either merged or single. All the strings are already quite long so any shortening that could help legibility is welcome.

```suggestion
            $info['notificationMessage'] = '<p>' . Piwik::translate('PrivacyManager_ConsentManagerDetected', [$templateData['consentManagerName'], '<a href="' . $templateData['consentManagerUrl'] . '" target="_blank" rel="noreferrer noopener">', '</a>']) . '</p>';
```

Reply:
changed as suggested